test/hotspot/jtreg/compiler/vectorapi/VectorGatherMaskFoldingTest.java

```diff
@@ -0,0 +1,897 @@
+    static {
+        for (int i = 0; i < L_SPECIES.length(); i++) {
+            longArray[i] = i + 1;
+            longArray2[i] = L_SPECIES.length() - i + 1;
+            longMask[i] = i % 2 == 0;
+            longMask2[i] = i >= L_SPECIES.length() / 2;
+            longOffsets[i] = (i + L_SPECIES.length() / 2) % L_SPECIES.length();
+            longOffsets2[i] = (L_SPECIES.length() - i) % L_SPECIES.length();
+        }
+        longVector = LongVector.fromArray(L_SPECIES, longArray, 0);
+        longVector2 = LongVector.fromArray(L_SPECIES, longArray2, 0);
+        longVectorMask = VectorMask.fromArray(L_SPECIES, longMask, 0);
+        longVectorMask2 = VectorMask.fromArray(L_SPECIES, longMask2, 0);
+        for (int i = 0; i < I_SPECIES.length(); i++) {
+            intArray[i] = i + 1;
+            intArray2[i] = I_SPECIES.length() - i + 1;
+            intMask[i] = i % 2 == 0;
+            intMask2[i] = i >= I_SPECIES.length() / 2;
+            intOffsets[i] = (i + I_SPECIES.length() / 2) % I_SPECIES.length();
+            intOffsets2[i] = (I_SPECIES.length() - i) % I_SPECIES.length();
+        }
+        intVector = IntVector.fromArray(I_SPECIES, intArray, 0);
+        intVector2 = IntVector.fromArray(I_SPECIES, intArray2, 0);
+
+        intVectorMask = VectorMask.fromArray(I_SPECIES, intMask, 0);
+        intVectorMask2 = VectorMask.fromArray(I_SPECIES, intMask2, 0);
+        for (int i = 0; i < D_SPECIES.length(); i++) {
+            doubleArray[i] = (double) i + 1.0;
+            doubleArray2[i] = (double) (D_SPECIES.length() - i) + 1.0;
+            doubleMask[i] = i % 2 == 0;
+            doubleMask2[i] = i >= D_SPECIES.length() / 2;
+            doubleOffsets[i] = (i + D_SPECIES.length() / 2) % D_SPECIES.length();
+            doubleOffsets2[i] = (D_SPECIES.length() - i) % D_SPECIES.length();
+        }
+        doubleVector = DoubleVector.fromArray(D_SPECIES, doubleArray, 0);
+        doubleVector2 = DoubleVector.fromArray(D_SPECIES, doubleArray2, 0);
+        doubleVectorMask = VectorMask.fromArray(D_SPECIES, doubleMask, 0);
+        doubleVectorMask2 = VectorMask.fromArray(D_SPECIES, doubleMask2, 0);
+        for (int i = 0; i < F_SPECIES.length(); i++) {
+            floatArray[i] = i + 1.0f;
+            floatArray2[i] = F_SPECIES.length() - i + 1.0f;
+            floatMask[i] = i % 2 == 0;
+            floatMask2[i] = i >= F_SPECIES.length() / 2;
+            floatOffsets[i] = (i + F_SPECIES.length() / 2) % F_SPECIES.length();
+            floatOffsets2[i] = (F_SPECIES.length() - i) % F_SPECIES.length();
+        }
+        floatVector = FloatVector.fromArray(F_SPECIES, floatArray, 0);
+        floatVector2 = FloatVector.fromArray(F_SPECIES, floatArray2, 0);
+        floatVectorMask = VectorMask.fromArray(F_SPECIES, floatMask, 0);
+        floatVectorMask2 = VectorMask.fromArray(F_SPECIES, floatMask2, 0);
+    }
+
+    // LOAD TESTS
+
+    // LongVector tests
+
+    @Test
+    @Warmup(10000)
```

Comment:
Since all tests use the same warmup, I would suggest to set it once via `testFrameworkobject.setDefaultWarmup(10000)`, see https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/compiler/lib/ir_framework/README.md

Reply:
👍 thanks for the suggestion!